presto-main/src/test/java/com/facebook/presto/operator/TestingBlockBuilders.java

```diff
@@ -0,0 +1,282 @@
+                MethodHandle keyNativeEquals = TYPE_MANAGER.resolveOperator(OperatorType.EQUAL, ImmutableList.of(keyType, keyType));
+                MethodHandle keyBlockNativeEquals = compose(keyNativeEquals, nativeValueGetter(keyType));
+                MethodHandle keyNativeHashCode = TYPE_MANAGER.resolveOperator(OperatorType.HASH_CODE, ImmutableList.of(keyType));
+                MethodHandle keyBlockHashCode = compose(keyNativeHashCode, nativeValueGetter(keyType));
+
+                block = fromKeyValueBlock(
+                        Optional.of(isNull),
+                        offsets,
+                        keyBlock,
+                        valueBlock,
+                        ((MapType) type),
+                        keyBlockNativeEquals,
+                        keyNativeHashCode,
+                        keyBlockHashCode);
+            }
+            else if (type instanceof RowType) {
+                List<Type> fieldTypes = ((RowType) type).getTypeParameters();
+                Block[] fieldBlocks = new Block[fieldTypes.size()];
+
+                for (int i = 0; i < fieldBlocks.length; i++) {
+                    Block fieldBlock = buildBlockWithType(
+                            fieldTypes.get(i),
+                            positionCount,
+                            allowNulls, isView,
+                            wrappings);
+                    fieldBlocks[i] = fieldBlock;
+                }
+
+                block = fromFieldBlocks(positionCount, Optional.of(isNull), fieldBlocks);
+            }
+            else {
+                throw new UnsupportedOperationException(format("type %s is not supported.", type));
+            }
+        }
+
+        if (isView) {
+            positionCount /= 2;
+            int offset = positionCount / 2;
+            block = block.getRegion(offset, positionCount);
+        }
+
+        if (wrappings.isPresent()) {
+            block = buildDictRleBlock(block, positionCount, wrappings);
+        }
+
+        return block;
+    }
+
+    static MapType createMapType(Type keyType, Type valueType)
+    {
+        MethodHandle keyNativeEquals = TYPE_MANAGER.resolveOperator(OperatorType.EQUAL, ImmutableList.of(keyType, keyType));
+        MethodHandle keyBlockNativeEquals = compose(keyNativeEquals, nativeValueGetter(keyType));
+        MethodHandle keyBlockEquals = compose(keyNativeEquals, nativeValueGetter(keyType), nativeValueGetter(keyType));
+        MethodHandle keyNativeHashCode = TYPE_MANAGER.resolveOperator(OperatorType.HASH_CODE, ImmutableList.of(keyType));
+        MethodHandle keyBlockHashCode = compose(keyNativeHashCode, nativeValueGetter(keyType));
+        return new MapType(
+                keyType,
+                valueType,
+                keyBlockNativeEquals,
+                keyBlockEquals,
+                keyNativeHashCode,
+                keyBlockHashCode);
+    }
+
+    private static String getRandomString(int maxStringLength)
```

Comment:
maxStringLength -> length (this methods returns string of exactly the size specified in this parameter)